src/solver/clause.rs

```diff
@@ -230,6 +234,63 @@ impl Clause {
         )
     }
 
+    fn conditional(
+        parent_id: VariableId,
+        requirement: Requirement,
+        condition: VersionSetId,
+        decision_tracker: &DecisionTracker,
+        requirement_candidates: impl IntoIterator<Item = VariableId>,
+        condition_candidates: impl IntoIterator<Item = VariableId>,
+    ) -> (Self, Option<[Literal; 2]>, bool) {
+        assert_ne!(decision_tracker.assigned_value(parent_id), Some(false));
+        let mut condition_candidates = condition_candidates.into_iter();
+        let mut requirement_candidates = requirement_candidates.into_iter();
+
+        // Check if we have any condition candidates
+        let Some(first_condition) = condition_candidates.next() else {
+            // No conditions means this is just an assertion
+            return (
+                Clause::Conditional(parent_id, condition, requirement),
+                None,
+                false,
+            );
+        };
+
+        // Try to find a condition candidate that is undecided or true
+        let condition_literal = std::iter::once(first_condition)
+            .chain(condition_candidates)
+            .find(|&id| {
+                let value = decision_tracker.assigned_value(id);
+                value.is_none() || value == Some(true)
+            });
+
+        // Try to find a requirement candidate that is undecided or true
+        let requirement_literal = requirement_candidates.find(|&id| {
+            let value = decision_tracker.assigned_value(id);
+            value.is_none() || value == Some(true)
+        });
+
+        match (condition_literal, requirement_literal) {
+            // Found valid literals - use them
+            (Some(cond), _) => (
+                Clause::Conditional(parent_id, condition, requirement),
+                Some([parent_id.negative(), cond.negative()]),
+                false,
+            ),
+            (None, Some(req)) => (
+                Clause::Conditional(parent_id, condition, requirement),
+                Some([parent_id.negative(), req.positive()]),
+                false,
+            ),
+            // No valid literals found - conflict case
+            (None, None) => (
+                Clause::Conditional(parent_id, condition, requirement),
+                Some([parent_id.negative(), first_condition.negative()]),
+                true,
+            ),
+        }
```

Comment:
You can map the two iterators to their respective literals, then chain them, and eval the _literals_ to find one you need.